### DISCUSSION TITLE
Is it possible to share the async loop?

### DISCUSSION THREAD

**Discussion by smojef** [zauberzeug/nicegui#326] [Q&A] [answered]
Instead of calling run(), can I share the main asyncio loop with another module?  For instance a network message receiver.

**Comment by rodja** [comment, marked as answer]
The event loop is started by [Uvicorn](https://www.uvicorn.org/). You would need to pass the existing loop into uvicorn on start as shown in https://stackoverflow.com/a/66275953/364388. In our custom FastAPI example the uvicorn call is made explicit

https://github.com/zauberzeug/nicegui/blob/d095e67e599226b3dacf25ada3fce74f24430913/examples/fastapi/main.py#L17

I think you can simply add the `loop=...` parameter and be ready to go.

**Reply by smojef** [reply]
Thanks!
